src/views/SignTransaction.vue

```diff
@@ -29,13 +29,16 @@ export default class SignTransaction extends Vue {
     @State private keyguardResult!: KSignTransactionResult | Error | null;
 
     public async created() {
-        if (this.$route.name === `sign-transaction-success`) return;
+        if (this.keyguardResult instanceof Error) {
+            this.rpcState.reply(ResponseStatus.ERROR, this.keyguardResult);
```

Comment:
Since the sign-transaction flow does not currently have a landing page in the AccountsManager, there is also no place to display an error or have the user try again. Thus we forward the error directly to the caller (the Safe, for example) which automatically closes the window from the caller side.

Reply:
Could also be a nice preliminary comment.